### PR TITLE
Skip tests incompatible with valgrind

### PR DESCRIPTION
**Pull request by ldorau**
Ref: #338

### CODE REVIEW

**Comment by ldorau** [comment]
The NIghtly build: https://github.com/ldorau/unified-memory-framework/actions/runs/8279332960/job/22653507638

**Comment by bratpiorka** [comment]
will this be a temporary solution? - I agree that we have to enable nightly but we should also investigate this more. Please add a TODO in the code :)

**Comment by ldorau** [comment]
> will this be a temporary solution? - I agree that we have to enable nightly but we should also investigate this more. Please add a TODO in the code :)

No, it is not a temporary solution, see https://github.com/oneapi-src/unified-memory-framework/issues/338

File: test/test_valgrind.sh [hunk 1]
```diff
@@ -79,7 +79,18 @@ for test in $(ls -1 ./umf_test-*); do
 	SUP="${WORKSPACE}/test/supp/${test}.supp"
 	OPT_SUP=""
 	[ -f ${SUP} ] && OPT_SUP="--suppressions=${SUP}"
-	HWLOC_CPUID_PATH=./cpuid valgrind $OPTION $OPT_SUP --gen-suppressions=all $test >$LOG 2>&1 || echo -n "(valgrind failed) "
+
+	# skip tests incompatible with valgrind
+	case $test in
+	./umf_test-memspace_host_all)
+		FILTER='--gtest_filter="-*allocsSpreadAcrossAllNumaNodes"'
```
Comment (lukaszstolarczuk) [inline test/test_valgrind.sh:86]: one thing to consider: do we have some functionality in these tests that still should be run under Valgrind? If so and we could split the test - we should split them.

Comment (ldorau) [reply test/test_valgrind.sh:86]: Yes, maybe ... It is an option to be considered.

Comment (ldorau) [reply test/test_valgrind.sh:86]: ... but not in this PR for sure :-D

Comment (bratpiorka) [reply test/test_valgrind.sh:86]: @ldorau please add an issue about that so we don't forget to do it at our free time

Comment (ldorau) [reply test/test_valgrind.sh:86]: Done: https://github.com/oneapi-src/unified-memory-framework/issues/347

**Pull request merged**